test/views/html/TaxCalculationViewSpec.scala

```diff
@@ -0,0 +1,341 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.html
+
+import models._
+import org.joda.time.LocalDate
+import org.jsoup.nodes.Document
+import views.html.cards.home.taxCalculation
+import models.UnderpaidStatus.{Unknown => UnderpaidUnknown, _}
+import models.OverpaidStatus.{Unknown => OverpaidUnknown, _}
+import util.{BetterOptionValues, LanguageHelper}
+import viewmodels.TaxCalculationViewModel
+
+class TaxCalculationViewSpec extends ViewSpec {
+
+  import BetterOptionValues._
+
+  def view(reconciliation: Reconciliation): Option[Document] =
+    TaxCalculationViewModel.fromTaxYearReconciliation(TaxYearReconciliation(2017, reconciliation))(config).map {
+      taxRec =>
+        asDocument(taxCalculation(taxRec)(messages, pertaxContext, config).toString)
+    }
+
+  def formatDate(date: LocalDate) = LanguageHelper.langUtils.Dates.formatDate(Some(date), "dd MMMM yyyy")(messages)
+
+  val underpaidUrlReasons: String =
```

Comment:
No need to construct these urls from scratch as they already exist in the config decorator. It will also make the system slightly more resilient to change as if a url changes it only needs to change in 1 place